src/cmap/wire_protocol/compression.ts

```diff
@@ -102,9 +103,10 @@ export function decompress(
       if (Snappy[PKG_VERSION].major <= 6) {
         Snappy.uncompress(compressedData, { asBuffer: true }, callback);
       } else {
-        Snappy.uncompress(compressedData, { asBuffer: true })
-          .then(buffer => callback(undefined, buffer))
-          .catch(error => callback(error));
+        Snappy.uncompress(compressedData, { asBuffer: true }).then(
+          buffer => callback(undefined, buffer),
+          error => callback(error)
```

Comment:
just double checking that this is what we want; the previous handler would have caught an error thrown inside the `.then` (i.e., the .then call to the callback); the new behavior would only handle the error returned from the original `uncompress`

Reply:
Yea, this is a common mistake when callbackifying a promise API. if the callback throws (which it shouldn't but nonetheless it could) then you call the callback with its own error, this avoids that.